hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/api/i2c_api.c

```diff
@@ -94,12 +97,19 @@ int i2c_read(i2c_t *obj, int address, char *data, int length, int stop) {
     I2C_Type *base = i2c_addrs[obj->instance];
     i2c_master_transfer_t master_xfer;
 
-    i2c_address = address;
+    i2c_address = address >> 1;
     memset(&master_xfer, 0, sizeof(master_xfer));
-    master_xfer.slaveAddress = address;
+    master_xfer.slaveAddress = address >> 1;
     master_xfer.direction = kI2C_Read;
     master_xfer.data = (uint8_t *)data;
     master_xfer.dataSize = length;
+    if (obj->next_repeated_start) {
```

Comment:
Is this repeated start flag needed ? Because the default ``master_xfer.flags`` (=0) is start + stop. The ``stop`` argument changes stop behavior. Therefore if ``stop`` is true, the ``i2c_read()`` here would do start, no stop. What am I missing?